databricks_cli/pipelines/cli.py

```diff
@@ -70,22 +70,25 @@ def deploy_cli(api_client, spec_arg, spec):
 
     databricks pipelines deploy --spec example.json
     """
+
     if bool(spec_arg) == bool(spec):
         raise RuntimeError('The spec should be provided either by an option or argument')
     src = spec_arg if bool(spec_arg) else spec
     spec_obj = _read_spec(src)
     if 'id' not in spec_obj:
-        pipeline_id = str(uuid.uuid4())
-        click.echo("Updating spec at {} with id: {}".format(src, pipeline_id))
-        spec_obj['id'] = pipeline_id
-        _write_spec(src, spec_obj)
-    _validate_pipeline_id(spec_obj['id'])
-    PipelinesApi(api_client).deploy(spec_obj)
+        response = PipelinesApi(api_client).create(copy.deepcopy(spec_obj))
+        new_pipeline_id = response['pipeline_id']
+        click.echo("Successfully deployed pipeline: {}".format(
+            _get_pipeline_url(api_client, new_pipeline_id)))
 
-    pipeline_id = spec_obj['id']
-    base_url = "{0.scheme}://{0.netloc}/".format(urlparse(api_client.url))
-    pipeline_url = urljoin(base_url, "#joblist/pipelines/{}".format(pipeline_id))
-    click.echo("Pipeline successfully deployed: {}".format(pipeline_url))
+        spec_obj['id'] = new_pipeline_id
+        _write_spec(src, spec_obj)
+        click.echo("Updated spec at {} with ID {}".format(src, new_pipeline_id))
```

Comment:
@anew, where did we land on if we're going to update the spec in place or not? In a Rahul's implementation, we always do, which is fine with me.